packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/Mediator.kt

```diff
@@ -19,11 +19,14 @@ package io.realm.kotlin.internal
 import io.realm.kotlin.types.BaseRealmObject
 import kotlin.reflect.KClass
 
-// TODO Public due to being a transative dependency of ConfigurationImpl
+// TODO Public due to being a transitive dependency of ConfigurationImpl
 public interface Mediator {
     // TODO OPTIMIZE Most usage of this could be done from cached RealmObjectCompanion instance.
     //  Maybe just eliminate this method to ensure that we don't misuse it in favor of
     //  companionOf(clazz).`io_realm_kotlin_newInstance`()
     public fun createInstanceOf(clazz: KClass<out BaseRealmObject>): RealmObjectInternal
+    public fun createInstanceOf(classSimpleName: String): RealmObjectInternal
     public fun companionOf(clazz: KClass<out BaseRealmObject>): RealmObjectCompanion
+    public fun companionOf(classSimpleName: String): RealmObjectCompanion
+    public fun getClassOrThrow(classSimpleName: String): KClass<out BaseRealmObject>
```

Comment:
Yeah, that's probably right. I added the `KClass` to the cache so that it can be retrieved it using `ClassInfo` derived from the `ClassKey`, which comes from the link itself.